### ISSUE TITLE
BCE loss mps device

### ISSUE THREAD

**Issue by mbelotti-nuk** [pytorch/pytorch#137001]
### 🐛 Describe the bug

I have a problem training a model with the BCE loss using mps device. The model uses a linear layer to predict a the sign of a real value number (two classes). When I train it in cpu everything is fine, but when i pass to mps the prediction degenerates and predicts always the same "class" for all the training points.

### Versions

Collecting environment information...
PyTorch version: 2.3.0.dev20240229
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.6.1 (arm64)
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.29.2
Libc version: N/A

Python version: 3.12.4 | packaged by conda-forge | (main, Jun 17 2024, 10:13:44) [Clang 16.0.6 ] (64-bit runtime)
Python platform: macOS-14.6.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M3 Max

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] torch==2.3.0.dev20240229
[pip3] torchaudio==2.2.0.dev20240229
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.18.0.dev20240229
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] torch                     2.3.0.dev20240229          pypi_0    pypi
[conda] torchaudio                2.2.0.dev20240229          pypi_0    pypi
[conda] torchsummary              1.5.1                    pypi_0    pypi
[conda] torchvision               0.18.0.dev20240229          pypi_0    pypi

**Comment by skotapati** [comment]
Hi @mbelotti-nuk, can you provide a script that can be used to reproduce this issue? Unfortunately without repro instructions I won't be able to debug further

**Comment by malfet** [comment]
Closing as no reproducer was provided — @skotapati asked for a script to reproduce this and none was given, with no comments in over a year. Please open a new issue with a self-contained repro if this still reproduces on a current PyTorch release.
